ui/apps/platform/src/Components/ReportJobsTable.tsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import { Bullseye, Button, Text } from '@patternfly/react-core';
+import {
+    ActionsColumn,
+    ExpandableRowContent,
+    Table,
+    Tbody,
+    Td,
+    Th,
+    Thead,
+    Tr,
+} from '@patternfly/react-table';
+
+import useSet from 'hooks/useSet';
+import useAuthStatus from 'hooks/useAuthStatus';
+import { ReportSnapshot } from 'services/ReportsService.types';
+import { saveFile } from 'services/DownloadService';
+import { getDateTime } from 'utils/dateUtils';
+import ReportJobStatus from 'Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobStatus';
+import { ComplianceScanSnapshot } from 'services/ComplianceScanConfigurationService';
+import EmptyStateTemplate from './EmptyStateTemplate';
+
+export type ReportJobsTableProps<T> = {
+    snapshots: T[];
+    getJobId: (data: T) => string;
+    getConfigName: (data: T) => string;
+    onClearFilters: () => void;
+    onDeleteDownload: (reportId) => void;
+    renderExpandableRowContent: (snapshot: T) => React.ReactNode;
+};
+
+type Snapshot = ReportSnapshot | ComplianceScanSnapshot;
+
+const filenameSanitizerRegex = new RegExp('(:)|(/)|(\\s)', 'gi');
+
+const onDownload = (snapshot: Snapshot, jobId: string, configName: string) => () => {
+    const { completedAt } = snapshot.reportStatus;
+    const filename = `${configName}-${completedAt}`;
+    const sanitizedFilename = filename.replaceAll(filenameSanitizerRegex, '_');
+    return saveFile({
+        method: 'get',
+        // @TODO: We may need to allow passing specific endpoints depending on backend
+        url: `/api/reports/jobs/download?id=${jobId}`,
+        data: null,
+        timeout: 300000,
+        name: `${sanitizedFilename}.zip`,
+    });
+};
+
+function ReportJobsTable<T extends Snapshot>({
+    snapshots,
+    getJobId,
+    getConfigName,
+    onClearFilters,
+    onDeleteDownload,
+    renderExpandableRowContent,
+}: ReportJobsTableProps<T>) {
+    const { currentUser } = useAuthStatus();
+    const expandedRowSet = useSet<string>();
+
+    return (
+        <Table aria-label="Jobs table" variant="compact">
+            <Thead>
+                <Tr>
+                    <Th>
+                        <span className="pf-v5-screen-reader">Row expansion</span>
+                    </Th>
+                    <Th width={25}>Completed</Th>
+                    <Th width={25}>Status</Th>
+                    <Th width={50}>Requestor</Th>
+                    <Th>
+                        <span className="pf-v5-screen-reader">Row actions</span>
+                    </Th>
+                </Tr>
+            </Thead>
+            {snapshots.length === 0 && (
+                <Tbody>
+                    <Tr>
+                        <Td colSpan={5}>
```

Comment:
Seeing `5` here brought question to mind whether it needs to be a variable?

Comparing to Vulnerability Reporting as baseline, there is conditional rendering depending on permissions:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/VulnReports/VulnReportsPage.tsx#L565-L572

Ouch, which might differ for Compliance Reports:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/VulnReports/VulnReportsPage.tsx#L80-L84

Reply:
Ah, sorry @pedrottimark did you confuse the Report Jobs table with the VM Reports table? The VM Reports equivalent should be located in apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobs.tsx. There should only be 5 columns (the last is the row actions).

Reply:
So we only allow you to `download a report` or `delete the generated download` if you were the one who initiated the job. There isn't any write access for deleting a download. @charmik-redhat @lvalerom Do you know if that is correct and is intended to behave that way?